Makefile

```diff
@@ -1,5 +1,15 @@
 # Set the binaries
 SHELL := /bin/bash
+
+ifeq '$(findstring ;,$(PATH))' ';'
```

Comment:
To detect the OS, you should be able to use `uname` command. In my case on my Linux, `uname` returns `Linux`, on Mac it should return `Darwin`, on Windows it should return `WindowsNT`.

Another option is to use the macro `OSTYPE`.

Reply:
`uname` is not generally available on Windows, and `OSTYPE` is an environment variable commonly available on shells, but not guaranteed.

The else condition does use `uname` for Linux and Darwin, but it attempts to detect Windows first as *nix systems use `:` for path separators, and windows uses `;` this trick allows it to quickly identify Windows.

In the event the user is using something like CYGWIN or MINGW ... these are also identified.

It's probably a bit much, TBH ... as my goal was to just get things working on a Mac, but laying the foundation for other possible platforms at this stage seemed like a good idea.